### DISCUSSION TITLE
converting multiple jpegs at once

### DISCUSSION THREAD

**Discussion by danman138** [ImageMagick/ImageMagick#4734] [Development]
on mac I am injecting a jpeg background into the transparent layer of a .png using terminal with ImageMagick 6.9.12-34
I use this and it works for 1 image at a time. I need to use it on every .png in folder 

Convert background multibackground.jpeg 22213-168.png -gravity center -compose over -composite “output/22213-186.png”

(Multibackground.png is the background I’m adding to the transparent layer)

(22213-168.png is the avatar with transparent background)

(“Output” is the folder it is sent to and 22213.168.png is the finished product with background 

For removing background I used:

convert -transparent “#638596” 22213.png 22213-168.png

22213.png was the original and 22213-168.png was to be new output with transparent background

To run every .png file in directory at once I ran:

convert -transparent "#8e6fb6" *.png "output/ *.png"

Using *.png to target all .png files

I try using this to inject all the backgrounds at once but only get 1 file that is partially done:

Convert background multibackground.jpeg *.png -gravity center -compose over -composite “output/ *.png”

**Comment by fmw42** [comment]
You cannot use wild cards for both input and output. Furthermore your commands have too many images specified or I do not understand the use of "background" after convert.

You can use mogrify to insert all images in a directory into a given common transparent background. Suppose all your images are in a folder called test1 on your desktop. Create a new directory to hold the results called test2 on the desktop. Then

cd
cd desktop/test1
mogrify -path ../test2 -format png -draw "image over 0,0 0,0 'background.png' " *.jpg

See https://legacy.imagemagick.org/Usage/basics/#mogrify_compose

Perhaps you might provide an example input and background and what you expect for the output.
